https.go

```diff
@@ -149,14 +149,25 @@ func (proxy *ProxyHttpServer) handleHttps(w http.ResponseWriter, r *http.Request
 			go copyAndClose(ctx, proxyClientTCP, targetTCP)
 		} else {
 			go func() {
+				var err error
 				var wg sync.WaitGroup
 				wg.Add(2)
-				go copyOrWarn(ctx, targetSiteCon, proxyClient, &wg)
-				go copyOrWarn(ctx, proxyClient, targetSiteCon, &wg)
+				go func() {
+					err = copyOrWarn(ctx, targetSiteCon, proxyClient)
+					wg.Done()
+				}()
+				go func() {
+					copyOrWarn(ctx, proxyClient, targetSiteCon)
```

Comment:
Your explanation in the PR description about why we're only checking the error in the previous `copyOrWarn()` call was very useful!  Could you put a short version of that in a comment here?

Reply:
Done! Thanks for calling that out, I'm not sure why I omitted it.